views/js/controller/Tools/assessmentActivity.js

```diff
@@ -100,88 +158,89 @@ define([
         $('.paused-assessments', $assessmentActivityContainer).text(data.paused_assessments);
     }
 
-    function updateActivityGraph(data) {
-        var $activityGraphContainer = $();
-
+    function updateActivityGraph(config) {
+        if (!activityGraph) {
+            activityGraph = activityGraphFactory({
+                autoRefresh: config.autoRefresh,
+                autoRefreshBar: true,
+                graphConfig: {
+                    bindto: config.bindto,
+                    data: {
+                        url: url.route('completedAssessmentsData', 'Tools', 'taoProctoring')
+                    },
+                    axis: {
+                        x: {
+                            label: {
+                                text: __('Hours'),
+                            }
+                        },
+                        y: {
+                            label: {
+                                text: __('Completed tests'),
+                            },
+                            tick: {format: d3.format("d")},
+                        }
+                    },
+                    tooltip: {
+                        format: {
+                            name: function () {
+                                return __('Completed');
+                            }
+                        }
+                    },
+                    legend: {
+                        show: false
+                    }
+                }
+            })
+            .render();
+        } else {
+            activityGraph.refresh({
+                graphConfig: {
+                    data: {
+                        url: url.route('completedAssessmentsData', 'Tools', 'taoProctoring', { interval: config.interval })
```

Comment:
It looks like every component still call its own data endpoint to refresh their data instead of being grouped as suggested.